### PR TITLE
Fixes and workarounds

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: src/library/scala/runtime/Tuple2Zipped.scala [hunk 1]
```diff
@@ -119,7 +119,7 @@ object Tuple2Zipped {
                 w2: T2 <:< CC2[El2],
                 bf: scala.collection.generic.CanBuildFrom[CC1[_], (El1, El2), That]
       ): That = {
-        val buf = bf(x._1)
+        val buf = bf(w1(x._1).asInstanceOf[CC1[_]])
```
Comment (DarkDimius) [inline src/library/scala/runtime/Tuple2Zipped.scala]: is this cast indeed needed?

Comment (nicolasstucki) [reply src/library/scala/runtime/Tuple2Zipped.scala]: The original error was
```scala
122 |        val buf = bf(x._1)
    |                     ^^^^
    |                     found:    T1
    |                     required: CC1[_]
    |           
```
and intermediate error is
```scala
122 |        val buf = bf(w1(x._1))
    |                     ^^^^^^^^
    |                     found:    CC1[El1]
    |                     required: CC1[_]
```

Comment (nicolasstucki) [reply src/library/scala/runtime/Tuple2Zipped.scala]: It looks like `CC1[El1]` is not a subtype `CC1[_]`. Is this an intended change? Or just a bug.

File: src/library/scala/sys/SystemProperties.scala [hunk 2]
```diff
@@ -61,7 +68,7 @@ object SystemProperties {
    */
   def exclusively[T](body: => T) = this synchronized body
 
-  implicit def systemPropertiesToCompanion(p: SystemProperties): SystemProperties.type = this
+  // implicit def systemPropertiesToCompanion(p: SystemProperties): SystemProperties.type = this
```
Comment (DarkDimius) [inline src/library/scala/sys/SystemProperties.scala]: I think this should be fixed in dotty.

**Comment by nicolasstucki** [comment]
updated

**Pull request closed without merging**